x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Avg.java

```diff
@@ -75,6 +80,7 @@ protected Expression.TypeResolution resolveType() {
 
     private Avg(StreamInput in) throws IOException {
         super(in);
+        pragmas = in.readNamedWriteable(QueryPragmas.class);
```

Comment:
All these need to be guarded by a TransportVersion check, see above.
Plus, you'll need the `writeTo` part.